tests/unit/curated_recommendations/test_rankers.py

```diff
@@ -341,3 +352,105 @@ def test_boost_preferred_topic_no_reorder(self):
         )
 
         assert recs == not_reordered_recs
+
+
+class TestCuratedRecommendationsProviderBoostFollowedSections:
+    """Unit tests for boost_followed_sections"""
+
+    @staticmethod
+    def generate_curated_recommendations_feed_sections(
+        received_feed_ranks: list[int], section_ids: list[str]
+    ) -> CuratedRecommendationsFeed:
+        """Create dummy sections & construct dummy CuratedRecommendationsFeed object for the tests below with
+        specific receivedFeedRank per section.
+        """
+        sections = []
+        layout_order = [layout_4_medium, layout_4_large, layout_6_tiles]
+        i = 1
+        for rank in received_feed_ranks:
+            section = Section(
+                receivedFeedRank=rank,
+                recommendations=[],
+                title="Fake Section Title " + str(i),
+                layout=layout_order[0],
+            )
+            sections.append(section)
+            i += 1
+
+        feed = CuratedRecommendationsFeed()
+        for section_id, section in zip(section_ids, sections):
+            setattr(feed, section_id, section)  # Dynamically set Section attributes
+
+        return feed
+
+    def test_boost_followed_sections(self):
+        """Test boosting sections works properly. Followed & unfollowed sections should
+        maintain their relative order.
+        """
+        req_sections = [
+            SectionConfiguration(sectionId="arts", isFollowed=True, isBlocked=False),
+            SectionConfiguration(sectionId="business", isFollowed=True, isBlocked=False),
+            SectionConfiguration(sectionId="travel", isFollowed=False, isBlocked=True),
+        ]
+        feed = self.generate_curated_recommendations_feed_sections(
+            [0, 5, 3, 2, 6], ["top_stories_section", "arts", "food", "business", "travel"]
```

Comment:
I think it would be more readable if the order that you provide the sections in determines the `receivedFeedRank`. Is there a reason to use a separate argument for `receivedFeedRank`?
```suggestion
            ["top_stories_section", "arts", "food", "business", "travel"]
```

Reply:
you mean just increment the rank for each section? I wanted it to be a bit more configurable for the tests